sycl/source/detail/queue_impl.cpp

```diff
@@ -62,6 +63,35 @@ static event createDiscardedEvent() {
 event queue_impl::memset(const std::shared_ptr<detail::queue_impl> &Self,
                          void *Ptr, int Value, size_t Count,
                          const std::vector<event> &DepEvents) {
+#if XPTI_ENABLE_INSTRUMENTATION
+  /// This section of code is relying on scoped objects, so they cannot be
+  /// encapsulated in a function
+  detail::tls_code_loc_t Tls;
```

Comment:
Are we going to use this for all new code or only if tracing is being added for the existing code to avoid ABI-breakage? If the former, then PR should be split into two - infrastructure + rewrite of all existing codes and the 2nd PR containing this new instrumentatio (IIUC). If the latter, then extra comments on that are needed near the class definition.

Reply:
Existing functionality doesn't have to be rewritten as we propagate code location information all the way to the SYCL object.   How we approach this moving forward will change for all new instrumentation. With so many gaps currently in the instrumentation, I would like to provide end users with new data now and address refactoring existing instrumentation at a later time.

Reply:
No, existing instrumentation doesn't have to be changed. We could refactor it at a later time. Current PRs are to address requested features from end users that cover instrumentation gaps.